model_api/python/openvino/model_api/models/classification.py

```diff
@@ -100,24 +110,71 @@ def parameters(cls):
                 "multilabel": BooleanValue(
                     default_value=False, description="Predict a set of labels per image"
                 ),
+                "hierarchical": BooleanValue(
+                    default_value=False,
+                    description="Predict a hierarchy if labels per image",
+                ),
+                "hierarchical_config": StringValue(
+                    default_value="",
+                    description="Extra config for decoding hierarchical predicitons",
+                ),
+                "confidence_threshold": NumericalValue(
+                    default_value=0.5, description="Predict a set of labels per image"
+                ),
             }
         )
         return parameters
 
     def postprocess(self, outputs, meta):
-        print(self.multilabel)
         if self.multilabel:
             return self.get_multilabel_predictions(
                 outputs[self.out_layer_names[0]].squeeze()
             )
+        elif self.hierarchical:
+            return self.get_hierarchical_predictions(
+                outputs[self.out_layer_names[0]].squeeze()
+            )
         return self.get_multiclass_predictions(outputs)
 
+    def get_hierarchical_predictions(self, logits: np.ndarray):
```

Comment:
Should there be greedy label resolver?

Reply:
At some point yes, it's just a basic pre-processing, as mentioned in PR's heading

Reply:
Added